### ISSUE TITLE
Using Docker Compose

### ISSUE THREAD

**Issue by jhallam3** [DanWahlin/AspNetCorePostgreSQLDockerApp#5]
Hi, When I follow both your Pluralsight training and your code. I cant seam to get it to build the docker images. 
Ive set the connection string to 
            var connectionString = "User ID=postgres;Password=password;Server=localhost;Port=5432;Database=postgres;Integrated Security=true;Pooling=true;";

then doing a docker build 

docker build -f core.development.dockerfile -t Username/core:latest .

it errors with 

An error occurred while calling method 'ConfigureServices' on startup class 'Core.Startup'. Consider using IDbContextFactory to o
verride the initialization of the DbContext at design-time. 

Error: Connection refused  
System.Net.Sockets.SocketException: Connection refused  
   at Npgsql.NpgsqlConnector.Connect(NpgsqlTimeout timeout)  
   at Npgsql.NpgsqlConnector.RawOpen(NpgsqlTimeout timeout)  
   at Npgsql.NpgsqlConnector.Open(NpgsqlTimeout timeout)  
   at Npgsql.ConnectorPool.Allocate(NpgsqlConnection conn, NpgsqlTimeout timeout)  
   at Npgsql.NpgsqlConnection.OpenInternal()  
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.Open()  
   at Microsoft.EntityFrameworkCore.Storage.Internal.NpgsqlDatabaseCreator.Exists()  
   at Microsoft.EntityFrameworkCore.Migrations.HistoryRepository.Exists()  
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration)  
   at Microsoft.EntityFrameworkCore.Design.MigrationsOperations.UpdateDatabase(String targetMigration, String contextType)  
   at Microsoft.EntityFrameworkCore.Tools.Cli.DatabaseUpdateCommand.<>c__DisplayClass0_0.<Configure>b__0()  
   at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)  
   at Microsoft.EntityFrameworkCore.Tools.Cli.Program.Main(String[] args)  
Connection refused      

the solution runs fine its now getting it to build in to docker. 

My docker file is below. 

FROM microsoft/dotnet:latest

MAINTAINER Jon Hallam

ENV DOTNET_USE_POLLING_FILE_WATCHER=1

COPY . /app
# COPY ./package.json /var/www/aspnetcoreapp/package.json

WORKDIR /app

RUN dotnet restore

RUN dotnet build

EXPOSE 5000/tcp

RUN ["dotnet", "ef", "database", "update"]
# ENTRYPOINT ["dotnet", "watch", "run", "--server.urls", "http://0.0.0.0:5000"]

ENTRYPOINT ["dotnet", "run", "--server.urls", "http://0.0.0.0:5000"]
# Build the image:
# docker build -f core.development.dockerfile -t username/core:latest .
# Option 1
# Start PostgreSQL and ASP.NET Core (link ASP.NET core to ProgreSQL container with legacy linking)
# docker run -d --name my-postgres -p 5432:5432 -e POSTGRES_PASSWORD=password postgres
# docker run -d -p 5000:5000 --link my-postgres:postgres username/core:latest
# Option 2: Create a custom bridge network and add containers into it
# docker network create --driver bridge isolated_network
# docker run -d --net=isolated_network --name postgres -p 5432:5432 -e POSTGRES_PASSWORD=password postgres
# docker run -d --net=isolated_network --name aspnetcoreapp -p 5000:5000 username/core:latest

basically I have Migrations that works great in dev however getting it to deploy migrations when its deployed to docker with postgres it fails.

**Comment by DanWahlin** [comment]
While you can individually build the images, I'm going to recommend using Docker Compose for that. That's why it's there - to simplify that process. The other option is there mainly to show that it's possible to build individual images, link them, etc. but I never use that option as it's not productive. Docker Compose handles it...have you tried that?

**Comment by jhallam3** [comment]
I use dock compose with the dockerfile you have as a example. That works
well.

I have it working now. It was migrations not applying on startup.

Also is there a way of having two connection string for different
environments? At the mo I change to localhost and Postgres in two locations
the app.setting and the applicationdbcontext. Just trying to be more
productive.  :)

Thanks for a great example having that has been a god send!



Regards

Jon Hallam


On 20 Oct 2016, at 17:22, Dan Wahlin <notifications@github.com> wrote:

While you can individually build the images, I'm going to recommend using
Docker Compose for that. That's why it's there - to simplify that process.
The other option is there mainly to show that it's possible to build
individual images, link them, etc. but I never use that option as it's not
productive. Docker Compose handles it...have you tried that?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/DanWahlin/AspNetCorePostgreSQLDockerApp/issues/5#issuecomment-255155274>,
or mute the thread
